### DISCUSSION TITLE
Make Auto Volume Growth Configurable

### DISCUSSION THREAD

**Discussion by alejandrorascovan** [seaweedfs/seaweedfs#6157] [Ideas]
Currently, the settings for automatic volume growth appear to be hard-coded in the project. It would be great to make these settings configurable externally.

For instance, in our use case, we work with relatively small volume sizes (around 1GB). If all volumes fill up in less than 14 minutes, which seems to be the current [interval](https://github.com/seaweedfs/seaweedfs/blob/master/weed/server/master_grpc_server_volume.go#L51) at which the auto volume growth routine runs, there would be no writable volumes available during this time window.

It would be great to have the ability to configure both the auto volume growth interval and how many volumes are added during each growth cycle.

**Comment by chrislusf** [comment]
@kmlebedev could you please help with this?

**Reply by kmlebedev** [reply]
Yes, there was an idea to calculate these intervals using a formula.
In fact, we know the volume size and at least the network speed and limits on the output.
I don't want to overcomplicate the settings and a step of two volumes at a time is reasonable for now
